src/main/java/chess/domain/game/GameStatus.java

```diff
@@ -0,0 +1,101 @@
+package chess.domain.game;
+
+import chess.domain.piece.Piece;
+import chess.domain.piece.PieceColor;
+import chess.domain.piece.PieceType;
+import chess.domain.square.File;
+import chess.domain.square.Square;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class GameStatus {
+
+    private static final int INITIAL_KING_COUNT = 2;
+    private static final double PENALTY_RATE = 0.5;
+    private static final String ERROR_NOT_EXIST_KING = "현재 보드에 남아 있는 킹이 없습니다.";
+
+    private final Map<Square, Piece> pieces;
+
+    public GameStatus(final Map<Square, Piece> pieces) {
+        this.pieces = pieces;
+    }
+
+    public double calculateTotalScore(final PieceColor color) {
+        return calculateScoreWithoutPawn(color) + calculatePawnScore(color);
+    }
+
+    private double calculateScoreWithoutPawn(final PieceColor color) {
+        return pieces.values().stream()
+                .filter(piece -> !piece.isSameType(PieceType.PAWN))
+                .filter(piece -> piece.isSameColor(color))
+                .mapToDouble(piece -> piece.getType().getScore())
```

Comment:
```suggestion
                .mapToDouble(Piece::getScore)
```
만들어주신 메소드로 대체가 가능해보이네요.

Reply:
그렇네요. 말씀해 주신 대로 변경하였습니다. 😸